### PR TITLE
ref: Refactor `photodiodeSpot` and `photodiodeGhostBox`

### LINKED ISSUE #322: Rename `photodiode_spot` as `pdSpotEncode` and add docstrings
The name is more akin to what's actually occurring here, plus a lot of the related code to the photodiode calls it this

### PR DESCRIPTION
**Pull request by RobertGemmaJr**
- `photodiodeGhostBox` is now exported as a constant instead of a function
- Renames `photodiodeSpot` as `pdSpotEncode`
- Adds docstrings

The original name for the photodiode function was `pdSpotEncode`. I'm returning to that name - I think it's a little more prescriptive and it'll be nice for users to have that consistent naming. Eventually we're going to turn it into a custom  jsPsych extension,  we can think about renaming then.

closes #322

### CODE REVIEW

File: src/lib/markup/photodiode.js [hunk 1]
```diff
@@ -4,29 +4,42 @@ import { eventCodes } from "../../config/trigger";
...
+  }
 
-  // Pulse the spot color from black to white
+  /**
+   * Pulses the photodiode spot from black (on) to white (off) and runs a callback function
+   * @param {number} ms The amount of time to flash the photodiode spot
+   * @param {function} callback A callback function to execute after the flash
+   */
   function pulseFor(ms, callback) {
     $(".photodiode-spot").css({ "background-color": "black" });
```
Comment (galenwinsor) [inline src/lib/markup/photodiode.js:42]: There's jQuery in this???

File: src/lib/markup/photodiode.js [hunk 2]
```diff
@@ -4,29 +4,42 @@ import { eventCodes } from "../../config/trigger";
...
-  const spot = span("", { id: "photodiode-spot", class: "photodiode-spot" });
-  return div(spot, {
-    id: "photodiode-box",
-    // Photodiode is only visible if config.USE_PHOTODIODE is true
-    class: config.USE_PHOTODIODE ? "photodiode-box visible" : "photodiode-box invisible",
-  });
-}
+const photodiodeGhostBox = div(span("", { id: "photodiode-spot", class: "photodiode-spot" }), {
+  id: "photodiode-box",
+  class: config.USE_PHOTODIODE ? "photodiode-box visible" : "photodiode-box invisible",
```
Comment (galenwinsor) [inline src/lib/markup/photodiode.js:14]: Is `config` guaranteed to be defined when this is run? It always makes me a little nervous to put conditional stuff at the top level.

File: src/lib/markup/photodiode.js (hunk 1 above)
Comment (RobertGemmaJr) [reply src/lib/markup/photodiode.js:42]: Oh yeah! I'm afraid to touch it 😨

File: src/lib/markup/photodiode.js (hunk 2 above)
Comment (RobertGemmaJr) [reply src/lib/markup/photodiode.js:14]: It is, but I agree I'd like to change it at some point.

**Comment by RobertGemmaJr** [comment]
Current dependencies on/for this PR:
* main
  * **PR #337** <a href="https://app.graphite.dev/github/pr/brown-ccv/honeycomb/337?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
    * **PR #353** <a href="https://app.graphite.dev/github/pr/brown-ccv/honeycomb/353?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #356** <a href="https://app.graphite.dev/github/pr/brown-ccv/honeycomb/356?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/brown-ccv/honeycomb/337?utm_source=stack-comment).

**Review by broarr — APPROVED** [review]

**Pull request merged**